ONSdigital/sdc-responses-dashboard
# Use npm for JS dependencies

### Motivation and Context
Having JS loading from HTML scripts was not ideal for many reasons such linting or testing.

### What has changed
- As a first pass, I have used npm to handle all the JS dependencies.
- Added eslint for JS files.
- Added prettier plugin to format js and work with eslint.
- Travis will now lint JS files.
- Broke the JS into much smaller pieces to make the process clearer.
- `getCollexFromSurveyId` now uses `Array.forEach` instead of standard for loop.
- Simplified and made it clearer the creation of the `counter`box elements. 
- Stuck to the pattern of using [JS function expressions](https://developer.mozilla.org/en-US/docs/web/JavaScript/Reference/Operators/function) over [function declerations](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function#Function_declaration_hoisting). 
---

Future plans:
- Use LESS compiler to move CSS dependencies to npm.


### How to test?
- App should still work as expected.
- Check the new commands in readme and make sure they work as expected.

### Links




### Checklist

* [x] STATIC_ASSETS_VERSION updated.
